portlist/poller.go

```diff
@@ -31,14 +31,15 @@ type Poller struct {
 
 	c chan List // unbuffered
 
+	initOnce sync.Once // guards init of private fields
```

Comment:
move this up a couple lines before `c`.  (normally mutexes and sync.once etc come before the stuff they guard)